build/run.js

```diff
@@ -432,16 +461,35 @@ async function runCommand(command,argv) {
     runner()
 }
 
+function get_target_platform(argv) {
+    let target = argv.target
+    if (target === undefined) {
+        const local_platform = os.platform()
+        switch (local_platform) {
+            case 'darwin':
+                return 'macos'
+            case 'win32':
+                return 'win'
+            default:
+                return local_platform
+        }
+    }
+    return target
+}
+
 async function main () {
+    let argv = optParser.parse()
+
+    const target = get_target_platform(argv)
```

Comment:
I dont think thsi is the responsibility of the main function to get the target. This should be built into the `updateBuildVersion` function instead.